metaflow/runner/subprocess_manager.py

```diff
@@ -10,23 +10,58 @@
 from typing import Callable, Dict, Iterator, List, Optional, Tuple
 
 
-def kill_process_and_descendants(pid, termination_timeout):
+def send_signals(pid, signal):
     # TODO: there's a race condition that new descendants might
     # spawn b/w the invocations of 'pkill' and 'kill'.
     # Needs to be fixed in future.
     try:
-        subprocess.check_call(["pkill", "-TERM", "-P", str(pid)])
-        subprocess.check_call(["kill", "-TERM", str(pid)])
+        subprocess.call(["pkill", signal, "-P", str(pid)])
```

Comment:
Not using check_call here on purpose, if the subprocess doesn't have any descendants, check_call raises an exception. This (for some arcane reason) slowed the code down substantially.